pykern/pkcli/github.py

```diff
@@ -326,6 +326,43 @@ def restore(git_txz):
         _shell(["git", "checkout"])
 
 
+def verify_head_passed_ci(repo):
+    import requests
+
+    branch = "master"  # TODO(rorour) find a way to set master/main
+    owner = "radiasoft"  # TODO(rorour) remove var
+
+    def _run_on_repo(*cmd):
+        return (
+            subprocess.check_output(
+                ["git", "-C", os.environ["HOME"] + f"/src/{owner}/{repo}", *cmd],
+                stderr=subprocess.STDOUT,
+            )
+            .decode("utf-8")
+            .rstrip("\n")
+        )
+
+    r = requests.get(
+        f"{_GITHUB_API}/repos/{owner}/{repo}/actions/runs?per_page=1&page=1&branch={branch}",
+        headers={
+            "Accept": "application/vnd.github+json",
+            "Authorization": "token " + os.environ["GITHUB_PAT"],
+        },
+    )
+    assert r.json().get("workflow_runs"), f"no workflow runs for branch {branch}"
+    api_head_sha = r.json().get("workflow_runs")[0].get("head_sha")
+    _run_on_repo("checkout", branch)
```

Comment:
The github3 documentation doesn't seem to be up to date on the web. I know this, because I read the code.
I found the [latest_sha](https://github.com/sigmavirus24/github3.py/blob/1df2461e830092413778d5f3ff2a8785821321cb/src/github3/repos/branch.py). Then I saw [check_runs](https://github.com/sigmavirus24/github3.py/blob/1df2461e830092413778d5f3ff2a8785821321cb/src/github3/repos/commit.py#L44) which does have a [conclusion](https://github.com/sigmavirus24/github3.py/blob/1df2461e830092413778d5f3ff2a8785821321cb/src/github3/checks.py#L208). 

I wrote this code in github.py which I think can do what you need which also reuses some of the code in github.py:

```py
def t(repo, branch=None):
    r = _repo_arg(repo)

    def _branch(r, name, reraise=True):
        try:
            return r.branch(name=name)
        except github3.exceptions.NotFoundError:
            if reraise:
                raise

    b = _branch(r, branch) if branch else _branch(r, "master", False) or _branch(r, "main")
    pkdp(b.latest_sha())
    pkdp([c.conclusion for c in b.commit.check_runs()])
```

Here's how it works:

```sh
$ pykern github t radiasoft/sirepo
../pykern/pykern/pkcli/github.py:131:t 8bf6d7c2fdf28c9979075d305d1988c4484a786a
../pykern/pykern/pkcli/github.py:132:t []
$ pykern github t radiasoft/sirepo --branch mantid
../pykern/pykern/pkcli/github.py:131:t 4d3d1322613c96a8b04d09f0443b2f1ef2550556
../pykern/pykern/pkcli/github.py:132:t ['success', 'success']
```

Reply:
b09f39dd081e33c2fea46f0f81697bb23d7c3e7f